### DISCUSSION TITLE
Action after Wakup from Deep Sleep

### DISCUSSION THREAD

**Discussion by Thorsten8509** [arendst/Tasmota#21734] [Q&A]
Hello@All,
i built a chicken coop door with a ESP32 based on Tasmota.
My problem ist that it needs to much power to use it on battery especcially during the winter.
So i need a solution to get my eps32 into deep sleep and wakeup after sunrise and sunset.
Every time it wakes up i would like to open the door at sunrise and close it on sunset.
I tried a few things but never got the solution for my problem.
Tasmota is configured as shutter. I hope anyone of you can help me and solve my problem.

**Comment by smhc** [comment]
You could do something like this:
https://github.com/smhc/catflap.esp32

From settings.txt
```
Rule2
on time#initialized do backlog ruletimer1 1 endon
on rules#timer=1 do ProcessWake endon
Rule2 1
```

The trick I used is to just check the time on boot and perform action 1 if greater than midday, otherwise action 2.

```
        var curmins   = (tasmota.time_dump(self.curtime)["hour"] * 60) + tasmota.time_dump(self.curtime)["min"]
        if (curmins > 720) # After midday
            mqtt.publish("catflap/lockstatus", string.format("locking %s", self.timestr), true)
            self.m.flapclose()
        else
            mqtt.publish("catflap/lockstatus", string.format("unlocking %s", self.timestr), true)
            self.m.flapopen()
        end
```

I assume you could write similar logic using rules rather than berry.

**Reply by Thorsten8509** [reply]
Thanks a lot for your suggestion but I didn't get this to work.
I found something more simple and I think this is possibly the right approach.

ON System#Boot DO Backlog event checksunrise=%time%; event checksunset=%time% ENDON ON event#checksunrise<%sunrise% DO Power1 1 ENDON ON event#checksunset>%sunset% DO Power2 1 ENDON"

I didn't get this to work yet. Maybe you ore somebody else could give a hint.

**Comment by reimerp** [comment]
I also failed with Deepsleep Timers and Rules. In the end you just have two times where a wakeup is needed, maybe check my script   https://github.com/arendst/Tasmota/discussions/22183 which is working but has to be adopted, since I need to wakeup during datetime each minute while sleeping the whole night. I did also some checks that battery lifetime with an M3 is much better than an esp32, i did not find a board without a 5V power IC. Also the Timer solution isn't sleeping the whole period, but just a max on one hour, at least this I noticed. The default DeepSleep does not have to wake up even when sleeping half a day. I have used an M3 on a MCP1700-3302 / 18650 Combination running in 5mins Intervals for ~4 month or more. However you might require more power for actuators like relais.... how is the door operated than? And the M3 has to be patched but this is rather easy to acconplish, especially when you compare the patch to an esp-01
